File: bank-service/src/main/java/rs/edu/raf/repository/CreditRequestRepository.java
package rs.edu.raf.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import rs.edu.raf.model.credit.CreditRequest;
import rs.edu.raf.model.credit.CreditRequestStatus;

import java.util.List;

public interface CreditRequestRepository extends JpaRepository<CreditRequest, Long> {
    List<CreditRequest> findAllByCreditRequestStatus(CreditRequestStatus creditRequestStatus);
    List<CreditRequest> findAllByBankAccount_OwnerAndCreditRequestStatus(Long owner, CreditRequestStatus creditRequestStatus);
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/repository/CreditRequestRepository.kt
package org.example.bankservicekotlin.repository

import org.example.bankservicekotlin.model.credit.CreditRequest
import org.example.bankservicekotlin.model.credit.CreditRequestStatus
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface CreditRequestRepository : JpaRepository<CreditRequest, Long> {
    fun findAllByCreditRequestStatus(creditRequestStatus: CreditRequestStatus): List<CreditRequest>
    fun findAllByBankAccount_OwnerAndCreditRequestStatus(
        owner: Long,
        creditRequestStatus: CreditRequestStatus
    ): List<CreditRequest>
}